cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -266,12 +266,24 @@ func PerformControlPlaneUpgrade(flags *applyFlags, client clientset.Interface, w
 		return DryRunStaticPodUpgrade(internalcfg)
 	}
 
+	// Don't save etcd backup directory if etcd is HA, as this could cause corruption
 	return PerformStaticPodUpgrade(client, waiter, internalcfg, flags.etcdUpgrade)
 }
 
+// GetPathManagerForUpgrade returns a path manager properly configured for the given MasterConfiguration.
+func GetPathManagerForUpgrade(internalcfg *kubeadmapi.MasterConfiguration, etcdUpgrade bool) (upgrade.StaticPodPathManager, error) {
+	isHAEtcd := false
+	if internalcfg.Etcd.External != nil && len(internalcfg.Etcd.External.Endpoints) > 1 {
```

Comment:
This logic can probably be a part of the existing etcd utils.  There are a bunch.